docs/src/dev/proof-aggregation.md

```diff
@@ -0,0 +1,190 @@
+        let mut stdin = ZKMStdin::new();
+        stdin.write(&30);
+        client.prove(&fibonacci_pk, stdin).compressed().run().expect("proving failed")
+    });
+
+    // Setup the inputs to the aggregation program.
+    let input_1 = AggregationInput { proof: proof_1, vk: fibonacci_vk.clone() };
+    let input_2 = AggregationInput { proof: proof_2, vk: fibonacci_vk.clone() };
+    let input_3 = AggregationInput { proof: proof_3, vk: fibonacci_vk.clone() };
+    let inputs = vec![input_1, input_2, input_3];
+
+    // Aggregate the proofs.
+    tracing::info_span!("aggregate the proofs").in_scope(|| {
+        let mut stdin = ZKMStdin::new();
+
+        // Write the verification keys.
+        let vkeys = inputs.iter().map(|input| input.vk.hash_u32()).collect::<Vec<_>>();
+        stdin.write::<Vec<[u32; 8]>>(&vkeys);
+
+        // Write the public values.
+        let public_values =
+            inputs.iter().map(|input| input.proof.public_values.to_vec()).collect::<Vec<_>>();
+        stdin.write::<Vec<Vec<u8>>>(&public_values);
+
+        // Write the proofs.
+        //
+        // Note: this data will not actually be read by the aggregation program, instead it will be
+        // witnessed by the prover during the recursive aggregation process inside Ziren itself.
+        for input in inputs {
+            let ZKMProof::Compressed(proof) = input.proof.proof else { panic!() };
+            stdin.write_proof(*proof, input.vk.vk);
+        }
+
+        // Generate the plonk bn254 proof.
+        client.prove(&aggregation_pk, stdin).plonk().run().expect("proving failed");
+    });
+}
+```
+
+In the host program, during the setup, the host will compile and load two guest programs as compiled ELF binaries: the `AGGREGATION_ELF` (representing the aggregation guest program) and `FIBONACCI_ELF`, whose corresponding guest program implementation can be found [here](https://github.com/ProjectZKM/Ziren/tree/main/examples/fibonacci). These programs are passed to `client.setup()` to generate the proving and verifying keys and to `client.prove()` to execute inside the zkVM and generate proofs. In this example, the host generates three compressed proofs proving the correct computation of Fibonacci for inputs n=10, 20, 30. 
+
+The host then prepares inputs corresponding to each Fibonacci proof for the aggregaton guest. Specifically, the host feeds the verification key hashes and raw public values as inputs and supplies the compressed proofs and full verification keys as witness data. The guest hashes the public values to a digest and calls `verify_zkm_proof(vk_hash, digest)` to check each proof. After all pass, the guest commits to the batch. That commitment becomes the public output of the aggregated proof.
+
+The aggregation program is ran inside the zkVM and generates a Plonk proof (representing the aggregated proof) that certifies the validity of the three individual Fibonacci proofs. 
```

Comment:
guest program

Reply:
We can add this: it also can be other proofs, like stark proof, or g16 proof. and this proof can be verified by zkm_verifier/wasm verifier.